frontend/src/metabase/query_builder/selectors.js

```diff
@@ -139,18 +139,25 @@ export const getFirstQueryResult = createSelector([getQueryResults], results =>
   Array.isArray(results) ? results[0] : null,
 );
 
+export const getTableId = createSelector([getCard], card =>
+  getIn(card, ["dataset_query", "query", "source-table"]),
+);
+
 export const getPKColumnIndex = createSelector(
-  [getFirstQueryResult],
-  result => {
+  [getFirstQueryResult, getTableId],
+  (result, tableId) => {
     if (!result) {
       return;
     }
     const { cols } = result.data;
-    const hasMultiplePks = cols.filter(isPK).length > 1;
+
+    const hasMultiplePks =
```

Comment:
One thing in the original code that at first took me in a not-ideal direction to solve this was that `hasMultiplePks` alone was not clear if it was about a question or a table.

Now it's sure clearer because we have the word `table` in `getIsPKFromTablePredicate` and its arg `tableId`.

It might me enough.

Or maybe we want to be even more clear that the whole check is also about a table and not a question by renaming `hasMultiplePks` something like `tableHasMultiplePks`

Maybe also renaming the selector `getPKColumnIndex` something like `getTablePKColumnIndex`.

I don't know if these things are already clear enough to most people, but thinking from the root cause of this issue, I'd think they weren't clear enough.